File: survival/src/main/java/vip/qoriginal/quantumplugin/registration/ReservedMinecraftRegistrationTest.java
package vip.qoriginal.quantumplugin.registration;

import org.bukkit.entity.Player;

public final class ReservedMinecraftRegistrationTest implements MinecraftRegistrationTest {
    @Override
    public boolean isAvailable() {
        return false;
    }

    @Override
    public StartResult start(Player player, Session session) {
        return StartResult.unavailable();
    }

    @Override
    public void cancel(Player player) {
        // Reserved for the future world-based registration test implementation.
    }
}

File: survival/src/main/java/vip/qoriginal/quantumplugin/registration/ReservedMinecraftRegistrationTest.kt
package vip.qoriginal.quantumplugin.registration

import org.bukkit.entity.Player

class ReservedMinecraftRegistrationTest : MinecraftRegistrationTest {
    override fun isAvailable(): Boolean = false

    override fun start(
        player: Player,
        session: MinecraftRegistrationTest.Session,
    ): MinecraftRegistrationTest.StartResult =
        MinecraftRegistrationTest.StartResult.unavailable()

    override fun cancel(player: Player) {
        // Reserved for the future world-based registration test implementation.
    }
}
